atlp-rwanda/cod-be
# Feature: user roles settings #181339527

**Name of the task**

User roles setting

**What does this PR do?**
The PR changes/assigns the role to users by the Super Administrator 
 
**Description of Task to be completed?**
Given that all users get registered with requester role, there is a need of a user with other privileges and permissions, this can be achieved through implementing this PR.
 
**How should this be manually tested?**

-  Login to your account with your email and password (Super Administrator) to get a valid token for authentication 
- Using Postman  add ```http://localhost:<port>/api/v1/users/assignRole ``` in the url 
- Pass the token in the authorization header as Bearer token  
- Do a post request with the data in the following format 
  {
    "rolename":"value",
    "user":"value" 
  } 
 rolename: The rolename as registered, check in migration file for role
 user: The user id to change his/her role

- The following are roles to be assigned to users
 
![image](https://user-images.githubusercontent.com/53344427/159998163-cafde49c-6a2b-4c00-9e2b-88cd17b933da.png)

Note that before testing you should run all migrations (npm run dbMigrate ) and seeds (npm run makeSeed), the super user admin and password can be found in src/database/seeders/20220321125534-super-admin.js file 

**Pivot Tracker ID**
#181414601
**Link to Pivot Tracker**
https://www.pivotaltracker.com/story/show/181414601

Endpoint
POST api/v1/users/assignRole
